### PR TITLE
Add example EIP-712 button to enable e2e testing

### PR DESCRIPTION
**Pull request by justin-lee-cb**

### CODE REVIEW

File: app/tip/[walletAddress]/page.tsx [hunk 1]
```diff
@@ -9,7 +9,12 @@ import { GeneratePaymentLink } from '@/app/util';
 import { useWallet } from '@/app/hooks/useWallet';
 import QRCode from 'qrcode';
 import QRCodeFooter from '@/app/component/qrCode';
+import { EXAMPLE_EIP_712_PAYLOAD } from '@/app/constants';
+
 import useRealtimeDb from '@/app/hooks/useRealtimeDb';
+import { generateEip712Payload } from '@/app/utils';
+
+const NFC_RELAYER_URL = 'https://nfc-relayer.vercel.app/api/paymentTxParams';
```
Comment (mykcryptodev) [inline app/tip/[walletAddress]/page.tsx]: We've got an env var for this

**Review by mykcryptodev — CHANGES_REQUESTED** [review]
looks good. using the NEXT_PUBLIC_NFC_RELAYER_URL env var consistently could be a good idea

**Review by mykcryptodev — APPROVED** [review]

File: app/tip/[walletAddress]/page.tsx [hunk 2]
```diff
@@ -126,6 +129,28 @@ export default function Tip() {
     }, 1000);
   };
 
+  const handleEip712TapToPay = useCallback(async () => {
+    // make a POST request to the NFC relayer
+    const response = await fetch(process.env.NEXT_PUBLIC_NFC_RELAYER_URL as string, {
```
Comment (amhed) [inline app/tip/[walletAddress]/page.tsx:134]: nit: let's wrap `process.env` variables into a global config object that casts to the correct type.

File: app/utils.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { EXAMPLE_EIP_712_PAYLOAD } from "./constants";
+
+export function generateEip712Payload() {
+  const nonceParams = generateNonceParams();
+  const eip712Payload = { ...EXAMPLE_EIP_712_PAYLOAD };
+  eip712Payload.rpcProxySubmissionParams.message.message.nonce = nonceParams.nonce;
```
Comment (amhed) [inline app/utils.ts:6]: why is this `message.message`?

File: app/utils.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  eip712Payload.rpcProxySubmissionParams.message.message.validBefore = String(nonceParams.validBefore);
+  
+  return eip712Payload;
+}
+
+/** Helper function that generates randomized params required for EIP712 signature */
+export function generateNonceParams() {
```
Comment (amhed) [inline app/utils.ts:13]: Exported functions should use jsDoc documents

File: app/utils.ts (hunk 3 above)
Comment (justin-lee-cb) [reply app/utils.ts:6]: The `message` field is supposed to be an `eth_signTypedData_v4` call which has the following type:

```
{
  types: { 
    ...
  },
  domain: {
    ...
  },
  primaryType: string;
  message: {
    ....
  }
}
````

Comment (amhed) [reply app/utils.ts:6]: That's... confusing 😅 , should it be named `signTypedData` instead?

**Pull request merged**